tests/wpunit/MediaItemQueriesTest.php

```diff
@@ -488,4 +487,87 @@ public function testQueryMediaItemBySourceUrl() {
 
     }
 
+	/**
+	 * testPostQuery
+	 *
+	 * This tests creating a small size media item and retrieving bigger size image via a GraphQL query
+	 *
+	 * @since 1.2.5
+	 */
+	public function testMediaItemNotExistingSizeQuery() {
+
+		/**
+		 * Create an attachment with a post set as it's parent
+		 */
+		$attachment_id = $this->createPostObject( [
+			'post_type'   => 'attachment',
+			'post_content' => 'some description',
+			'post_mime_type' => 'image/jpeg',
+		] );
+
+		$meta_data = [
+			'width' => 300,
+			'height' => 300,
+			'file' => 'example.jpg',
+			'sizes' => [
+				'thumbnail' => [
+					'file' => 'example-thumbnail.jpg',
+					'width' => 150,
+					'height' => 150,
+					'mime-type' => 'image/jpeg',
+				],
+			],
+			'image_meta' => [
+				'aperture' => 0,
+				'credit' => '',
+				'camera' => '',
+				'caption' => '',
+				'created_timestamp' => 0,
+				'copyright' => '',
+				'focal_length' => 0,
+				'iso' => 0,
+				'shutter_speed' => 0,
+				'title' => '',
+				'orientation' => '1',
+				'keywords' => [],
+			],
+		];
+
+		update_post_meta( $attachment_id, '_wp_attachment_metadata', $meta_data );
+		update_post_meta( $attachment_id, '_wp_attached_file', 'example.jpg' );
```

Comment:
I believe we should be able to change this to: 

```php
$filename      = ( WPGRAPHQL_PLUGIN_DIR . '/tests/_data/images/test.png' );
$attachment_id = $this->factory()->attachment->create_upload_object( $filename );
```

This will create the image upload with associated meta in the database the way WordPress does so that we're using "actual" data as WordPress would create it and not fully mocking the data. 

I think the change will break the tests where you're asserting string matches though.

Reply:
@jasonbahl Thanks, I updated the test code. Please check :)